app/helpers/application_helper/button/ems_network.rb

```diff
@@ -1,5 +1,6 @@
 class ApplicationHelper::Button::EmsNetwork < ApplicationHelper::Button::Basic
   def visible?
-    ::Settings.product.nuage
+    return ::Settings.product.nuage if @record.nil? # add new
+    @record.class == ManageIQ::Providers::Nuage::NetworkManager # edit existing
```

Comment:
We want to avoid constructs like the above as much as possible and use the `SupportsFeature` mixin instead.

In this case, I'd suggest something like:
```ruby
@record.supports?(:ems_network_new)
```

Reply:
Turns out `:ems_network_new` is even already present on [Nuage network manager](https://github.com/ManageIQ/manageiq-providers-nuage/blob/master/app/models/manageiq/providers/nuage/network_manager.rb#L13) so I could just literally replace the line with your suggestion and it works, thanks!

I think it's good enough if we use `:ems_network_new` even to determine if NetworkManager can be _edited_ (not just created) because the two operations are usually paired. Please let me know if you'd prefer to introduce a separate `:ems_network_edit` , but note that changes in core repo will be needed then.

Reply:
I think what matters here is the semantics of `:ems_network_new`: if you can create a new network you can also edit existing network. So I agree that you don't need to introduce `:ems_network_edit`.